scripts/crash-monitor

```diff
@@ -0,0 +1,393 @@
+#!/usr/bin/python
+
+import os
+import re
+import sys
+import syslog
+import logging
+import logging.handlers
+import commands
+import threading
+import traceback
+import redis
+import time
+import json
+import subprocess
+import swsssdk
+from optparse import OptionParser
+from swsssdk import SysmonDBConnector
+from swsssdk import _get_redis_dbid_port
+from time import gmtime, strftime
+
+reload(sys)
+sys.setdefaultencoding('utf-8')
+
+MAX_LOG_BYTES = 1024 * 1024
+PID_FILE = "/var/run/crash-monitor"
+CRASH_MONITOR_TABLE = "CRASH_MONITOR"
+PROC_EVENT_TABLE = "PROC_EVENT"
+
+STATE_WARM_STARTED = 'started'
+STATE_WARM_DONE = 'done'
+
+LOG_EMERG = 0
+LOG_ALERT = 1
+LOG_CRIT = 2
+LOG_ERR = 3
+LOG_WARNING = 4
+LOG_NOTICE = 5
+LOG_INFO = 6
+LOG_DEBUG = 7
+
+# Delay 0 seconds before trying auto-restart
+RESTART_DELAY_TIME = 0
+# Limit 3 restart within 1200 seconds
+RESTART_MAX_RETRIES = 3
+RESTART_CHECK_INTERVAL = 1200
+
+def setup_logger():
+    # Log File name
+    log_file = "/var/log/monitor-crash.log"
+    log = logging.getLogger('MONITOR_CRASH')
+    log.setLevel(logging.DEBUG)
+    # Create file log handler and set level to debug
+    fh = logging.handlers.RotatingFileHandler(log_file, maxBytes=(MAX_LOG_BYTES / 2), backupCount=1)
```

Comment:
Why use `logger` and a dedicated file which you rotate manually rather than just using the `syslog` library?

Reply:
Hi Joe,

thanks for help reviewing the change. 

This logger is for our internal use purpose. Will change it to syslog.

Thanks,
Haiyang

Reply:
> Why use `logger` and a dedicated file which you rotate manually rather than just using the `syslog` library?

Have updated the change by using `syslog` lib instead of `logger`.

Thanks,
Haiyang